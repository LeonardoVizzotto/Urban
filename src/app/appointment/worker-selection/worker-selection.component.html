<app-container *ngIf="availableWorkers$ | async; let workers">
  <span
    >Please chose the your preffered worker to meet you at
    {{ slot?.localisedTime }}</span
  >
  <app-card
    *ngFor="let worker of workers"
    [imageHidden]="true"
    [imageUrl]="worker.profilePicture?.thumbnail"
    (onClick)="selectWorker(worker)"
  >
    <span>{{ worker.name }} </span>
    <app-rating [rating]="worker.rating"></app-rating>
  </app-card>
</app-container>
