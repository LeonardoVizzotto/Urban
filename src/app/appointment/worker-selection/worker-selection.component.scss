span {
  text-align: center;
  width: 80%;
}
