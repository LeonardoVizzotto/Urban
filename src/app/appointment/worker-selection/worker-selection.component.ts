import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { Observable } from 'rxjs';
import { ROUTES } from 'src/app/routes.constants';
import { BasketService } from 'src/app/services/basket/basket.service';
import { Slot } from 'src/app/services/slots/model';
import { Worker } from 'src/app/services/workers/model';
import { WorkersService } from 'src/app/services/workers/workers.service';

@Component({
  selector: 'app-worker-selection',
  templateUrl: './worker-selection.component.html',
  styleUrls: ['./worker-selection.component.scss'],
})
export class WorkerSelectionComponent implements OnInit {
  public slot?: Slot;
  public availableWorkers$: Observable<Worker[]>;
  constructor(
    private basketService: BasketService,
    private workersService: WorkersService,
    private router: Router
  ) {}

  ngOnInit(): void {
    this.slot = this.basketService.selectedSlot;
    if (!this.slot) {
      this.router.navigate([ROUTES.SLOTS]);
      return;
    }

    this.availableWorkers$ = this.workersService.getAvailableWorkersDataBySlot(
      this.slot!.id
    );
  }

  public selectWorker(worker: Worker) {
    this.basketService.selectWorker(worker);
    this.router.navigate([ROUTES.CHECKOUT]);
  }
}
