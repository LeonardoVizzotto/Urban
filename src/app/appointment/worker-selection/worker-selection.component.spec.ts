import { ComponentFixture, TestBed } from '@angular/core/testing';

import { WorkerSelectionComponent } from './worker-selection.component';

describe('WorkerSelectionComponent', () => {
  let component: WorkerSelectionComponent;
  let fixture: ComponentFixture<WorkerSelectionComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ WorkerSelectionComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WorkerSelectionComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
