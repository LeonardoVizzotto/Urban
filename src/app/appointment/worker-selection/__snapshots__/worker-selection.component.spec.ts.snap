// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerSelectionComponent should match snapshot if a slot has been selected 1`] = `
<app-worker-selection
  availableWorkers$={[Function Observable]}
  basketService={[Function BasketService]}
  router={[Function Object]}
  slot={[Function Object]}
  workersService={[Function Object]}
>
  <app-container>
    <span>
      Please chose the your preffered worker to meet you at 
    </span>
    <app-card>
      <button
        class="card"
      >
        <img
          aria-hidden="true"
          src="https://randomuser.me/api/portraits/thumb/men/22.jpg"
        />
        <span>
          John Doe 
        </span>
        <app-rating>
          <div
            aria-label="Rating of this worker is 2.9 out of 5."
            class="rating"
            style="--rating: 2.9;"
          />
        </app-rating>
      </button>
    </app-card>
  </app-container>
</app-worker-selection>
`;
